src/Compilers/VisualBasic/Portable/Symbols/Attributes/SourceAttributeData.vb

```diff
@@ -233,6 +237,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                             foundMatch = parameterType = lazySystemType
                             k += 1
 
+                        Case CByte(SignatureTypeCode.SZArray)
+                            ' skip over and check the next byte
+                            foundMatch = parameterType.IsArrayType
+
                         Case Else
                             Return -1
                     End Select
```

Comment:
This is a large method, mostly language independent. It should be shared with C#, perhaps as a generic method parameterized by `Symbol` types.